### PR TITLE
Pass hyper params from client

### PR DESCRIPTION
**Pull request by SamuelGabriel**
### Change Description

**Brief** (a few bullet points describing your changes, use full sentences and try to link lines in the code whenever needed)
In this PR, I start passing all arguments to the server. To make this simpler I removed the local tabpfn functionality for now but kept the main code s.t. we can easily re-enable it later on. This was simpler, because the local tabpfn has very different parameters from the server one.

DEPENDENCY: this PR depends on https://github.com/automl/tabpfn-server/pull/48/ to filter out the "model" kwarg

### Standard Qs (leave questions that do not apply blank)
**If you broke behavior: Please describe what behavior you broke and how you inform people to not get stuck trying to use the old behavior.**

yes, i removed local

**If you used new dependencies: Did you add them to `requirements.txt`?**

no

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**
@liam-sbhoo @anshulg954 @davidotte 

---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

### CODE REVIEW

**Review by liam-sbhoo — COMMENTED** [review]
Looking great in overall!

Just some comments/questions:
- regarding the model args, why not pop the `model` args in the client instead and keep the server untouched? Then we'll have one less change to make when we have the feature enabled.
- can we add some test cases, concerning the presence of `tabpfn_config`? Just a very minimal safe guard to ensure that we take `tabpfn_config`

**Comment by SamuelGabriel** [comment]
Tests: yeah, good point. I will add a test ensuring the defaults are send.

"model" arg: at some point we will need this model arg anyways, as we will train new models and want users to be able to choose, so I thought why not now. I should make it backward compatible on the backend, though.

File: tabpfn_client/prompt_agent.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def prompt_and_set_token(cls, user_auth_handler: "UserAuthenticationClient"):
         # Registration
         if choice == "1":
             # validation_link = input(cls.indent("Please enter your secret code: "))
-            validation_link = "tabpfn-2023"
+            validation_link = "tabpfn-test"
```
Comment (liam-sbhoo) [inline tabpfn_client/prompt_agent.py:54]: The production is still using `tabpfn-2023`, right?

Comment (liam-sbhoo) [reply tabpfn_client/prompt_agent.py:54]: In the server, should we just keep the `validation_link` the same for all `DEV`, `PREPROD`, and `PROD`?
Or actually, do we still need this `validation_link` feature?

**Review by liam-sbhoo — COMMENTED** [review]
Some issues on the `validation_link`.

File: tabpfn_client/prompt_agent.py (hunk 1 above)
Comment (SamuelGabriel) [reply tabpfn_client/prompt_agent.py:54]: let's keep it synced. i will update it to be 2023 in DEV too

**Pull request merged**